aepsych/database/tables.py

```diff
@@ -85,10 +88,70 @@ def __repr__(self):
     @staticmethod
     def update(engine):
         logger.info("DBMasterTable : update called")
+        if not DBMasterTable._has_extra_metadata(engine):
+            DBMasterTable._add_extra_metadata(engine)
+        if not DBMasterTable._has_participant_id(engine):
+            DBMasterTable._add_participant_id(engine)
 
     @staticmethod
     def requires_update(engine):
-        return False
+        return not DBMasterTable._has_extra_metadata(
+            engine
+        ) or not DBMasterTable._has_participant_id(engine)
+
+    @staticmethod
+    def _has_extra_metadata(engine):
+        result = engine.execute(
+            "SELECT COUNT(*) FROM pragma_table_info('master') WHERE name='extra_metadata'"
+        )
+        rows = result.fetchall()
+        count = rows[0][0]
+        return count != 0
+
+    @staticmethod
+    def _has_participant_id(engine):
+        result = engine.execute(
+            "SELECT COUNT(*) FROM pragma_table_info('master') WHERE name='participant_id'"
+        )
+        rows = result.fetchall()
+        count = rows[0][0]
+        return count != 0
+
+    @staticmethod
+    def _add_participant_id(engine):
+        try:
+            result = engine.execute(
```

Comment:
This code is redundant with `_has_participant_id`. Do you even need to run these checks since you already call `_has_participant_id` on line 93?

Reply:
Considering participant_id column will be added at the same time as the metadata column, you are right in that it would be able to infer that if there is no participant_id column, there is no metadata column. But what about (for edge cases should this ever be a situation) if there was a participant_id column but no metadata column? Shouldn't we try to take into account any potential edge cases like those?

Reply:
Actually I see what you mean. I should just have one method that checks for a column name based on an input. But I still run the checks individually of each column to handle any extra edge cases